_tools/release_docs/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	// Finds markdown links of the form [foo](bar "alt-text").
+	linkRE = regexp.MustCompile(`\[([^]]*)\]\(([^)]*)\)`)
+	// Splits the link target into link target and alt-text.
+	altTextRE = regexp.MustCompile(`(.*)( ".*")`)
+
+	version = flag.String("version", "", "A version tag to process docs. (e.g. 1.0).")
+	remote  = flag.String("remote", "upstream", "The name of the remote repo from which to pull docs.")
+)
+
+func fixURL(u *url.URL) (modified bool) {
+	if u.Host != "" {
+		return
+	}
+
+	if strings.HasSuffix(u.Path, ".md") {
+		u.Path = u.Path[:len(u.Path)-3] + ".html"
+		modified = true
+	}
+
+	if strings.HasSuffix(u.Path, "/") {
+		u.Path = u.Path + "README.html"
+		modified = true
+	}
+	return
+}
+
+func processFile(filename string) error {
+	fileBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	output := linkRE.ReplaceAllFunc(fileBytes, func(in []byte) (out []byte) {
+		match := linkRE.FindSubmatch(in)
+
+		visibleText := string(match[1])
+		linkText := string(match[2])
+		altText := ""
+
+		if parts := altTextRE.FindStringSubmatch(linkText); parts != nil {
+			linkText = parts[1]
+			altText = parts[2]
+		}
+
+		u, err := url.Parse(linkText)
+		if err != nil {
```

Comment:
If it's not a valid link, the changes we do wouldn't matter anyway. To me this is just another way we don't touch the link like below where !fixURL.